Library/Homebrew/extend/os/linux/keg_relocate.rb

```diff
@@ -0,0 +1,78 @@
+class Keg
+  def relocate_dynamic_linkage(relocation)
+    # Patching patchelf using itself fails with "Text file busy" or SIGBUS.
+    return if name == "patchelf"
+    elf_files.each do |file|
+      file.ensure_writable do
+        change_rpath(file, relocation.old_prefix, relocation.new_prefix)
+      end
+    end
+  end
+
+  def change_rpath(file, old_prefix, new_prefix)
+    return if !file.elf? || !file.dynamic_elf?
+
+    patchelf = DevelopmentTools.locate "patchelf"
+    cmd_rpath = [patchelf, "--print-rpath", file]
+    old_rpath = Utils.popen_read(*cmd_rpath, err: :out).strip
+    # patchelf requires that the ELF file have a .dynstr section.
+    # Skip ELF files that do not have a .dynstr section.
+    return if ["cannot find section .dynstr", "strange: no string table"].include?(old_rpath)
+    raise ErrorDuringExecution, "#{cmd_rpath}\n#{old_rpath}" unless $CHILD_STATUS.success?
```

Comment:
Newline after the `raise`.  I also wonder whether the `ErrorDuringExecution ... unless $CHILD_STATUS.success?` could be a helper function in here?

Reply:
I factored out `Utils.safe_popen_read`, which is typically more useful. This case of inspecting the output of stderr is atypical.